CRM/Contribute/BAO/ContributionSoft.php

```diff
@@ -53,39 +53,17 @@ public static function add(&$params) {
    * Process the soft contribution and/or link to personal campaign page.
    *
    * @param array $params
-   * @param object $contribution CRM_Contribute_DAO_Contribution
+   * @param CRM_Contribute_BAO_Contribution $contribution
    *
+   * @throws \CRM_Core_Exception
+   * @throws \CiviCRM_API3_Exception
    */
   public static function processSoftContribution($params, $contribution) {
-    //retrieve existing soft-credit and pcp id(s) if any against $contribution
-    $softIDs = self::getSoftCreditIds($contribution->id);
```

Comment:
@colemanw any chance you can look at this PR - it really just moves these 2 lines ^^ inside the if blocks below (so we don't do the query if we don't care about the answer)